tokio/src/runtime/time/mod.rs

```diff
@@ -91,8 +93,11 @@ pub(crate) struct Driver {
 
 /// Timer state shared between `Driver`, `Handle`, and `Registration`.
 struct Inner {
-    // The state is split like this so `Handle` can access `is_shutdown` without locking the mutex
-    pub(super) state: Mutex<InnerState>,
+    /// The earliest time at which we promise to wake up without unparking.
+    next_wake: AtomicU64,
```

Comment:
Perhaps it would result in clearer code to wrap this in an `AtomicOptionNonZeroU64` utility type? Or at least we should simplify `NonZeroU64::new(t).unwrap_or_else(|| NonZeroU64::new(1).unwrap())`.

Reply:
Yes, it shoud be simplified. I have added a new helper type exactly called `AtomicOptionNonZeroU64`.

Compared to `unwrap_or_else` and `unwarp`, I prefer `match` here.